spec/realtime/failure.test.js

```diff
@@ -381,5 +381,56 @@ define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
 		});
 	}});
 
+	exports.no_messages_if_not_attached = function(test) {
+		
+		var testName = 'no_messages_if_not_attached';
+		var testMessage = { foo: 'bar', count: 1, status: 'active' };
+		var testMessage2 = { foo: 'bar', count: 2, status: 'active' };
+		
+		test.expect(2);
+		
+		try {
+
+			var sender_realtime = helper.AblyRealtime();
+			var sender_channel = sender_realtime.channels.get(testName);
+
+			var messageReceived = false;
+			
+			sender_channel.subscribe(function(message) {
+
+				if(messageReceived) {
+					test.ok(false, 'Message received when channel not in ATTACHED state.');
+				}
+
+				messageReceived = true;
+				test.ok(JSON.stringify(testMessage) === JSON.stringify(message.data), 'Check first message received');
```

Comment:
Equality assertion should use `test.equal(a, b, 'additional info')` rather than `test.ok(a === b, 'additional info')`, so that if it fails you get an `Expected 'foo' to equal 'bar'` rather than the somewhat less helpful `Expected false to be true`

Reply:
(you can also use test.deepEqual to avoid the need to json-stringify before comparing)

Reply:
https://github.com/ably/ably-js/pull/638/commits/abfce6fa43db763a3b92b2013a96c58294d75569